release/nightly_tests/dataset/map_batches_benchmark.py

```diff
@@ -147,6 +147,58 @@ def run_map_batches_benchmark(benchmark: Benchmark):
                 num_calls=1,
             )
 
+    # Test map_batches whose output is large and will be split into multiple blocks.
+    # With the following configuration, the total data size is (num_output_blocks * 1GB).
+    num_output_blocks = [10, 100, 1000]
+    input_size = 1024 * 1024
+    batch_size = 1024
+    ray.data.DataContext.get_current().target_max_block_size = 1024 * 1024
+    input_ds = (
+        ray.data.range(input_size).repartition(input_size // batch_size).materialize()
+    )
+
+    def map_batches_fn(num_output_blocks, batch):
+        per_row_output_size = (
+            ray.data.DataContext.get_current().target_max_block_size // len(batch["id"])
+        )
+        for _ in range(num_output_blocks):
+            yield {
+                "data": [
+                    np.zeros(shape=(per_row_output_size,), dtype=np.int8)
+                    for _ in range(len(batch["id"]))
+                ]
+            }
+
+    for num_blocks in num_output_blocks:
+        test_name = f"map-batches-multiple-output-blocks-{num_blocks}"
+
+        def test_fn():
+            ds = input_ds.map_batches(
+                lambda batch: map_batches_fn(num_blocks, batch),
+                batch_size=batch_size,
+                batch_format="numpy",
+            )
+            ds = ds.map_batches(
+                lambda batch: {"data_size": [sum(x.nbytes for x in batch["data"])]},
+                batch_size=batch_size,
+                batch_format="numpy",
+            )
+            ds = ds.materialize()
+            total_size = ds.sum("data_size")
+            expected_total_size = (
+                input_size
+                * num_blocks
+                * ray.data.DataContext.get_current().target_max_block_size
+                // batch_size
```

Comment:
I think we can just remove `batch_size` from this benchmark? It should be orthogonal. We can create the benchmark we need with just `map` or `map_batches` with a batch size of 1 right?

Reply:
Right, in theory, that should be the same. but I think setting a batch_size won't harm either. and I wanted to make it closer to a real case.